tests/integration/security/authn/jwt/jwt_test.go

```diff
@@ -0,0 +1,235 @@
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/util/file"
+	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/pkg/test/util/tmpl"
+	"istio.io/istio/tests/integration/security/util"
+	"istio.io/istio/tests/integration/security/util/connection"
+)
+
+const (
+	authHeaderKey = "Authorization"
+	validJwt      = "eyJhbGciOiJSUzI1NiIsImtpZCI6IkRIRmJwb0lVcXJZOHQyenBBMnFYZkNtcjVWTzVaRX" +
+		"I0UnpIVV8tZW52dlEiLCJ0eXAiOiJKV1QifQ.eyJleHAiOjM3NTU5ODU5MzQ2LCJpYXQiOjE1NTk4NTkzNDYsImlzcyI" +
+		"6InRlc3RpbmdAc2VjdXJlLmlzdGlvLmlvIiwic3ViIjoidGVzdGluZ0BzZWN1cmUuaXN0aW8uaW8ifQ.A4j9ft49KrKw" +
+		"zvgpzoKXdMxVyYTUjuU3LMgqjvfZXDSUIYtL6gx31CSMb2UZiESmJ7Xu8XkqhcZvmfcuU_WApl01emBOwjwtg50M-Yuc" +
+		"oy6A4DbnsVcKnSY7VtAWrJ9ACdr1CUPseTfkbaIW-PYDfLPTMJljO2NdzF0yC9l97N8lau9afNu9Ilc9DIlQ5CgW6h1J" +
+		"VasV21wSLlrglnWhut6yV6yQOqUjMcwgntgVkv_3UbdtzJKt46u29Juh_8m4OEdAXeyy00midPSIJWIm9J6lgLF35XUB" +
+		"e_nvonKwdsxKpV-LrWlKMZB8QneHbb3194J7Wr2H7WVnNGcbVUCORw"
+	expiredJwt = "eyJhbGciOiJSUzI1NiIsImtpZCI6IkRIRmJwb0lVcXJZOHQyenBBMnFYZkNtcjVWTzVaRXI0UnpIVV8tZ" +
+		"W52dlEiLCJ0eXAiOiJKV1QifQ.eyJleHAiOjE1NTk5MzEyMDksImlhdCI6MTU1OTkzMTIwOSwiaXNzIjoidGVzdGluZ0" +
+		"BzZWN1cmUuaXN0aW8uaW8iLCJzdWIiOiJ0ZXN0aW5nQHNlY3VyZS5pc3Rpby5pbyJ9.a7zuyoF7eaVcnHOkJKK25WqTd" +
+		"K4OT0sgQzrAqYhANv-6MfbIHgRtbOYvQW0pSL5saeSkpd8YV35NdmipsbPcMTVYgSlSPkmNdinZwJyGpqdGEu6fVYq3P" +
+		"FET0bBSm5yVTkO7yyX8AgVVH31ouGJ8OQ11gGZ66Jmle5PNwyGCh1ccZsT8LefYTbcDHMbXnoYwU4e3WwcphLBqFoafF" +
+		"JUpRXV5dtn2YLnMwA0ALTteRMVMYIkXQkR6QhBwUufC3aUQmZydzrGMaKqbwcYbcp1GG05v4A99rikNQ-Ia6xswgAEJu" +
+		"JaYhppL-0B7E-i4jhpGbKOrFyc6vFZlk9oejUpi7Q"
+)
+
+func TestAuthnJwt(t *testing.T) {
+	framework.NewTest(t).
+		Run(func(ctx framework.TestContext) {
+			ns := namespace.NewOrFail(t, ctx, "authn-jwt", true)
+
+			var a, b, c echo.Instance
+			echoboot.NewBuilderOrFail(ctx, ctx).
+				With(&a, util.EchoConfig("a", ns, false, nil, g, p)).
+				With(&b, util.EchoConfig("b", ns, false, nil, g, p)).
+				With(&c, util.EchoConfig("c", ns, false, nil, g, p)).
+				BuildOrFail(t)
+
+			testCases := []struct {
+				configFile string
+				subTests   []connection.Checker
+			}{
+				{
+					configFile: "simple-jwt-policy.yaml.tmpl",
+					subTests: []connection.Checker{
+						{
+							From: a,
+							Options: echo.CallOptions{
+								Target:   b,
+								PortName: "http",
+								Scheme:   scheme.HTTP,
+								Headers: map[string][]string{
+									authHeaderKey: {"Bearer " + validJwt},
+								},
+							},
+							ExpectSuccess: true,
+						},
+						{
+							From: a,
+							Options: echo.CallOptions{
+								Target:   b,
+								PortName: "http",
+								Scheme:   scheme.HTTP,
+								Headers: map[string][]string{
+									authHeaderKey: {"Bearer " + expiredJwt},
+								},
+							},
+							ExpectSuccess: false,
```

Comment:
are we able to get the respcode(200/401) and verify ?

Reply:
Good point. The checking function can check for success (200), but failure does not mean 401. I forgot to update that function :D